MediaPlaceholder multiple and gallery properties

LuigiPulcini opened the pull request:
The `gallery` property should be set to `true` only after checking that the selected `type` of media is not `audio`.

## Description
The MediaPlaceholder component can be used to select multiple files also with audio files. In addition to being counterintuitive, forcing `gallery` to be `true` only based on the `multiple` property does not allow to work with audio files. In fact, when `type` is set to `"audio"` and `multiple` is `true`, the MediaUpload component only shows images

## Additional Information
In addition to the proposed change, it could be possible to add another line as follows:

```
playlist={ multiple && type === "audio" }
```

This could be done when a playlist block is finally added to Gutenberg.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

LuigiPulcini committed:
MediaPlaceholder multiple property

The `gallery` property should be set to `true` only after checking that the selected `type` of media is not `audio`.

youknowriad commented on packages/editor/src/components/media-placeholder/index.js:
```diff
@@ -127,7 +127,7 @@ class MediaPlaceholder extends Component {
 				</FormFileUpload>
 				<MediaUpload
-					gallery={ multiple }
+					gallery={ multiple && type != "audio" }
```
I'd say, it's probably better if the check is `multiple && type === 'image'`

LuigiPulcini replied:
Yes, I agree.

And, actually, there should be two different lines, one for `gallery` and one for `playlist`, like this.

```
gallery={ multiple && type === 'image' }
playlist={ multiple && type === 'audio' }
```

LuigiPulcini committed:
gallery and playlist check for multiple and type

gallery and playlist check for multiple and type, being gallery true when multiple is true and type is image, while playlist true when multiple is true and type is audio.

youknowriad commented on packages/editor/src/components/media-placeholder/index.js:
```diff
@@ -127,7 +127,8 @@ class MediaPlaceholder extends Component {
 				<MediaUpload
-					gallery={ multiple }
+					gallery={ multiple && type === "image" }
+					playlist={ multiple && type === "audio" }
```
I don't think we support playlists yet in `MediaUpload` and even if we did, there are playlists of videos.

I think it's probably better to remove until #9169

LuigiPulcini replied:
I hear what you are saying regarding the video playlists and that could be easily changed to something accommodating for both audio and video types.

But I have to disagree regarding the possibility to wait until #9169: analyzing that code – particularly [this part of the edit.js file](https://github.com/WordPress/gutenberg/pull/9169/files#diff-7776436ea4fc43d5443c66f7d10ca453R102) – it is clear that it simply tried to circumvent the current MediaPlaceholder limitation, recreating the whole component inside the edit.js file, only because the current MediaPlaceholder implementation does not allow to manage media types other than images.

So, I believe it is important to fix the MediaPlaceholder before refactoring the playlist block, which will allow a more streamlined playlist component as well.

youknowriad replied:
The underlying `MediaUpload` component don't support the `playlist` yet. So why to pass a useless prop?

LuigiPulcini replied:
You are right.
I edited the proposed change accordingly.

My point is the `MediaPlaceholder` should be fixed before refactoring the playlist because of the reasons I explained above.

LuigiPulcini committed:
type check when setting the gallery property

The `gallery` property is set to `true` when `multiple` is `true` and `type === "image"`

youknowriad commented on packages/editor/src/components/media-placeholder/index.js:
```diff
@@ -127,7 +127,7 @@ class MediaPlaceholder extends Component {
 				</FormFileUpload>
 				<MediaUpload
-					gallery={ multiple }
+					gallery={ multiple && type === "image" }
```
Looks like there's an eslint issue here. We should use single quotes. You can test locally using `npm run lint` or install an eslint plugin for your IDE.

LuigiPulcini committed:
double to single quotes

changed type check using single quotes instead of double quotes

youknowriad reviewed (approved):
LGTM 👍
